### DISCUSSION TITLE
`SQLAlchemyBaseAccessTokenTable` does not exist in fastapi-users v9.2

### DISCUSSION THREAD

**Discussion by BracketJohn** [fastapi-users/fastapi-users#845] [Q&A] [answered]
## Describe the bug

I tried to follow the guide to set up authentication with a Cookie + Database backend. In [this step of the guide](https://fastapi-users.github.io/fastapi-users/configuration/authentication/strategies/database/), the guide tells me to import `SQLAlchemyBaseAccessTokenTable`, however, this does not exist anywhere (also using the github search it only shows up in the `docs/` folder.

## To Reproduce

1. Attempt the import as stated in the docs:
```py
from fastapi_users_db_sqlalchemy.access_token import (
    SQLAlchemyAccessTokenDatabase
)
```

It will fail on fastapu-users v9.2.x and v9.1.x (the two versions I tried).

## Expected behavior

The import does not fail.

## Additional context

Following the guide to setup `transport=cookie` and `strategy=database` proved to be hard in other ways, e.g.:
- [the Create a backend page](https://fastapi-users.github.io/fastapi-users/configuration/authentication/backend/) did not make clear enough to me, that I need to use my previously configured transport and strategy settings, instead giving the bearer + jwt example without any context 
- everything is `async`, is this on purpose? Many applications scale via starting multiple (docker-)instances instead of/in addition to the async approach

These two complexities may also have caused me to do something wrong at some point, so I'm sorry if it turns out that `SQLAlchemyBaseAccessTokenTable` does exist!

Thanks for the library, have a good night 😴

**Comment by frankie567** [comment, marked as answer]
`SQLAlchemyAccessTokenDatabase` is a class exposed from `fastapi_users_db_sqlalchemy`, which is a sub-dependency providing the SQLAlchemy classes: https://github.com/fastapi-users/fastapi-users-db-sqlalchemy It should be automatically installed when you do `pip install fastapi-users[sqlalchemy2]`.

If you want to make sure of it, you can do `pip freeze` and check that `fastapi_users_db_sqlalchemy` appears in your dependencies. If that's not the case, you can force to reinstall an up-to-date version with:

```
pip install -U fastapi-users[sqlalchemy2]
```

Concerning async, yes, this is definitely on purpose. FastAPI is an ASGI framework and clearly encourages the async approach. That said, it doesn't prevent us to spawn multiple workers, on the contrary! Async I/O helps us to better use the resource of a single worker by not blocking it on each I/O call.

**Reply by BracketJohn** [reply]
I see - we're using sqlalchemy 1.x so when testing I installed `fastapi-users[sqlalchemy]`, this then is the source of the problem! 

Can we use `fastapi-users[sqlalchemy2]` together with an additional install of `sqlalchemy@1.x`?

Thanks for checking it!

**Reply by frankie567** [reply]
It's called `sqlalchemy2`, but it's actually SQLAlchemy 1.4+ under the hood, so I don't think it would cause any issue.
